verilog/analysis/symbol_table.cc

```diff
@@ -455,6 +459,29 @@ class SymbolTable::Builder : public TreeContextVisitor {
     }
   }
 
+  void DescendEnumType(const SyntaxTreeNode& enum_type) {
```

Comment:
Hi @fangism,
Can you take a look on that?

Reply:
Exactly, this mirrors `DescendStructType`.

Reply:
naive question as bystander: if this is similar to DescentStructType, could this be made mostly one `static void DescendComplexType(specifics...)` and then have `DescentStructType() { DecentComplexType("struct", NodeEnum::kStruct, ..); }` and `DescentEnumType() { DescentComplexType("enum", NodeEnum::kEnum, ...); }` of sorts ?

Reply:
The difference between `struct` and `enum` is that we can access enum constants without referencing enum scope, e.g.
```
enum { a, b, c } e;
assign e = a;
```
vs
```
struct { int a; int b; } s;
assign x = s.a;
assign s.a = x;
```
To achive that enum constants have to be visible in an enum scope.

We can achive that in two ways:
1. Ignore enum scope and insert them directly in the scope - but we lose an information about enum (and its members)
2. Create enum similary as anonymous structure (which is already done) and additionaly insert enumeration constants in enum parent scope - question is what type they should be (inserting 'em as regular variables is a duplication)